ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -141,117 +115,76 @@ def vector_decoder(field_index, key):
-
-            repeated (bool): Boolean that represents if the Parameter is repeated or not.
-
-        Raises
-        ------
-            Exception: If the specified type is not supported.
-
-        Returns
-        -------
-            Callable[[int], Callable]: ScalarDecoder or VectorDecoder.
-
-        """
-        if type not in Context._FIELD_TYPE_TO_DECODER_MAPPING:
-            raise Exception(f"Error can not decode type '{type}'")
-        scalar, array = Context._FIELD_TYPE_TO_DECODER_MAPPING[type]
-        if repeated:
-            return array
-        return scalar
-
-    @staticmethod
-    def get_type_default(type: type_pb2.Field.Kind.ValueType, repeated: bool) -> Any:
-        """Get the Type default.
-
-        Args
-        ----
-            type (type_pb2.Field.Kind.ValueType): Type of the Parameter.
-
-            repeated (bool): Boolean that represents if the Parameter is repeated or not.
-
-        Returns
-        -------
-            Any: Default value.
-
-        """
-        if repeated:
-            return list()
-        type_default_value = Context._TYPE_DEFAULT_MAPPING.get(type)
-        return type_default_value
+_FIELD_TYPE_TO_ENCODER_MAPPING = {
+    type_pb2.Field.TYPE_FLOAT: (FloatEncoder, FloatArrayEncoder),
+    type_pb2.Field.TYPE_DOUBLE: (DoubleEncoder, DoubleArrayEncoder),
+    type_pb2.Field.TYPE_INT32: (IntEncoder, IntArrayEncoder),
+    type_pb2.Field.TYPE_INT64: (IntEncoder, IntArrayEncoder),
+    type_pb2.Field.TYPE_UINT32: (UIntEncoder, UIntArrayEncoder),
+    type_pb2.Field.TYPE_UINT64: (UIntEncoder, UIntArrayEncoder),
+    type_pb2.Field.TYPE_BOOL: (BoolEncoder, BoolArrayEncoder),
+    type_pb2.Field.TYPE_STRING: (StringEncoder, StringArrayEncoder),
+    type_pb2.Field.TYPE_ENUM: (IntEncoder, IntArrayEncoder),
+}
+
+_FIELD_TYPE_TO_DECODER_MAPPING = {
+    type_pb2.Field.TYPE_FLOAT: (FloatDecoder, FloatArrayDecoder),
+    type_pb2.Field.TYPE_DOUBLE: (DoubleDecoder, DoubleArrayDecoder),
+    type_pb2.Field.TYPE_INT32: (Int32Decoder, Int32ArrayDecoder),
+    type_pb2.Field.TYPE_INT64: (Int64Decoder, Int64ArrayDecoder),
+    type_pb2.Field.TYPE_UINT32: (UInt32Decoder, UInt32ArrayDecoder),
+    type_pb2.Field.TYPE_UINT64: (UInt64Decoder, UInt64ArrayDecoder),
+    type_pb2.Field.TYPE_BOOL: (BoolDecoder, BoolArrayDecoder),
+    type_pb2.Field.TYPE_STRING: (StringDecoder, StringArrayDecoder),
+    type_pb2.Field.TYPE_ENUM: (Int32Decoder, Int32ArrayDecoder),
+}
+
+_TYPE_DEFAULT_MAPPING = {
+    type_pb2.Field.TYPE_FLOAT: float(),
+    type_pb2.Field.TYPE_DOUBLE: float(),
+    type_pb2.Field.TYPE_INT32: int(),
+    type_pb2.Field.TYPE_INT64: int(),
+    type_pb2.Field.TYPE_UINT32: int(),
+    type_pb2.Field.TYPE_UINT64: int(),
+    type_pb2.Field.TYPE_BOOL: bool(),
+    type_pb2.Field.TYPE_STRING: str(),
+    type_pb2.Field.TYPE_ENUM: int(),
+}
+
+
+def get_encoder(
+    type: type_pb2.Field.Kind.ValueType, repeated: bool
+) -> Callable[[int], InnerEncoder]:
```

Comment:
```suggestion
) -> EncoderFactory:
```

Reply:
Changed this to what I'm calling `PartialEncoderConstructor`